README.md

```diff
@@ -15,6 +15,96 @@ gradient-free optimizers, and constrained optimization.
 * Armadillo: http://arma.sourceforge.net
 * OpenBLAS or Intel MKL or LAPACK (see Armadillo site for details)
 
+### Install Dependencies
+
+<details open>
+<summary>Linux</summary>
+
+Use your distributions' package manager to install the required dependencies. The commands shown below are for Ubuntu.
+
+```bash
+$ sudo apt-get update
+$ sudo apt-get install -y g++ cmake
```

Comment:
Not sure we should use `-y` here, a user might want to control the install process.

Reply:
@zoq, should I keep g++ and cmake installation as a part of the dependencies?
I think it should be removed (as it expected that the user has these installed).

Reply:
Agrred, I think we can use: http://mlpack.org/doc/mlpack-3.2.2/doxygen/build.html as orientation.

Reply:
The simple build instructions on the page don't install any deps, however, the detailed build instructions do. We, presently, in **ensmallen**, have the dependency installation as a separate subsection. Will I remove it?